### PR TITLE
update amazon-lambda-archetype to provide SAM templates

### LINKED ISSUE #5037: Amazon Lambda SAM CLI integration
**Description**
I've tested the SAM integration that is on the Amazon Lambda HTTP docs, and used a similar sam.jvm.yaml (without the dynamo db YAML), and it works fine.

Any chance the archetype and documentation could clone this functionality from the Amazon Lambda HTTP implementation (incl. uber jar) -- particularly whilst it's fresh in the minds of the developers.

```
./extensions/amazon-lambda-http/maven-archetype/target/classes/archetype-resources/sam.jvm.yaml
./extensions/amazon-lambda-http/maven-archetype/target/classes/archetype-resources/sam.native.yaml
./extensions/amazon-lambda-http/maven-archetype/src/main/resources/archetype-resources/sam.jvm.yaml
./extensions/amazon-lambda-http/maven-archetype/src/main/resources/archetype-resources/sam.native.yaml

```

```
AWSTemplateFormatVersion: '2010-09-09'
Transform: AWS::Serverless-2016-10-31

Globals:
  Function:
    Timeout: 5
Resources:
  QuarkusSam:
    Type: AWS::Serverless::Function
    Properties:
      Handler: io.quarkus.amazon.lambda.runtime.QuarkusStreamHandler::handleRequest
      Runtime: java8
      CodeUri: target/lambda-kotlin-1.0-SNAPSHOT-runner.jar
      MemorySize: 128
      Policies: AWSLambdaBasicExecutionRole

```

It would be super cool also to have this in Gradle which is my preferred build tool.

### PR DESCRIPTION
**Pull request by evanchooly**
merge helper scripts to one file and parameterize it to reduce clutter and repetition

fixes #5037

### CODE REVIEW

**Review by gastaldi — COMMENTED** [review]
That looks good to me, but I'll wait until someone with more experience in AWS to approve ;)

**Comment by geoand** [comment]
I am grossly underqualified to review this since I don't know much of anything about amazon lambda :)

**Comment by evanchooly** [comment]
it's mostly just the textual changes (not strictly amazon-related) that I wanted a newbie's eye on.  And the two of you reviewed the last round so i figured I'd give it another shot.  @oztimpower has verified the template changes, fwiw.

**Comment by geoand** [comment]
Sure, I'll go over the text, no problem

File: docs/src/main/asciidoc/amazon-lambda.adoc [hunk 1]
```diff
@@ -106,147 +104,116 @@ This will compile and package your code.
...
+[source]
+----
+LAMBDA_ROLE_ARN="arn:aws:iam::1234567890:role/lambda-role"
+----
 
 == Create the function
 
-The `create.sh` script is for deploying your lambda using the AWS Lambda Java runtime.
+The `manage.sh` script is for managing your lambda using the AWS Lambda Java runtime.  This script is provided only for
+your convenience.  Should you choose to Amazon provided tools, you can freely skip these sections and consult the
```
Comment (geoand) [inline docs/src/main/asciidoc/amazon-lambda.adoc]: `you choose to` -> `you choose to use` perhaps?

File: docs/src/main/asciidoc/amazon-lambda.adoc [hunk 2]
```diff
@@ -106,147 +104,116 @@ This will compile and package your code.
...
-Custom Runtime.
-
-
-== Create an Execution Role
-
-View the https://docs.aws.amazon.com/lambda/latest/dg/gettingstarted-awscli.html[Getting Started Guide] for deploying
-a lambda with AWS CLI.  Specifically make sure you have created an `Execution Role`.  You will need the `Role Arn` to copy/paste
-into the scripts generated by the maven archetype.
+The example lambda takes input passed in via the `--payload` switch which points to a json file
+in the root directory of the project.  If you want to see the return output of the lambda, open the `response.txt` file.
```
Comment (geoand) [inline docs/src/main/asciidoc/amazon-lambda.adoc]: `return output` maybe better as just plain `output`?

File: docs/src/main/asciidoc/amazon-lambda.adoc [hunk 3]
```diff
@@ -106,147 +104,116 @@ This will compile and package your code.
...
 
-.update-native.sh
 [source, subs=attributes+]
 ----
-aws lambda update-function-code --function-name my-native-function \
-           --zip-file fileb://target/function.zip
+sh manage.sh native create
 ----
 
+As above, commands can be stacked as needed.  The only requirement is that `native` be the first parameter should you wish
```
Comment (geoand) [inline docs/src/main/asciidoc/amazon-lambda.adoc]: `as needed` seems redundant

File: docs/src/main/asciidoc/amazon-lambda.adoc [hunk 4]
```diff
@@ -259,75 +226,18 @@ NB: With gradle, to build the uber-jar execute: ./gradlew quarkusBuild --uber-ja
...
-RUNNER=$( find . -maxdepth 1 -name '*-runner' )
-if [[ ! -z "$RUNNER" ]]
-then
-    $RUNNER
-fi
+sam local invoke --template sam.jvm.yaml --event {json test file}
 ----
 
-To invoke the native `provided` runtime with the `sam.native.yaml` template, use the following command:
+The native executable can also be locally tested the `sam.native.yaml` template:
```
Comment (geoand) [inline docs/src/main/asciidoc/amazon-lambda.adoc]: This sentence doesn't seem right

Comment (evanchooly) [reply docs/src/main/asciidoc/amazon-lambda.adoc]: updated s/executable/image/.  better?

Comment (geoand) [reply docs/src/main/asciidoc/amazon-lambda.adoc]: Now I see what I think is missing: `tested the sam.native.yaml` I assume should be `tested using the sam.native.yaml`

Comment (gsmet) [reply docs/src/main/asciidoc/amazon-lambda.adoc]: FTR, `native executable` is the wording we agreed on.

**Comment by gsmet** [comment]
@evanchooly can you squash everything once you're done? Thanks!

**Comment by evanchooly** [comment]
@gsmet for sure.

**Review by geoand — APPROVED** [review]
Docs look good!

**Pull request merged**